### ISSUE TITLE
[DOC] Full installation guide for H2O LLM Studio on Windows using WSL2

### ISSUE THREAD

**Issue by pascal-pfeiffer** [h2oai/h2o-llmstudio#290]
### 📃 Documentation

Let's add a start to finish guide so install H2O LLM Studio on Windows using WSL2.

### Motivation

Some links from the documentation are not what you need in WSL2. e.g. CUDA version should be WSL2 version.
https://developer.nvidia.com/cuda-downloads?target_os=Linux&target_arch=x86_64&Distribution=WSL-Ubuntu&target_version=2.0

**Comment by jenochs** [comment]
Draft Outline:
 Windows Operating System (Windows 10/11, etc)
 Hardware/Cloud Instance as applicable
 Windows Installation Elements
    -OS Specific Components
    -Drivers
    -WSL2 Installation
      -WSL2 Ubuntu Initial Configuration
            --Prerequisites, Permissions, etc
      -WSL2 Nvidia Cuda Drivers
      -WSL2 Python Installation
      -WSL2 Virtual Environment Setup
           -LLMStudio Virtual Environment Activation
           -Git Clone LLM Studio Repo
           -Navigating Repo
           -Requirements.txt
           -wave run
       -Navigating to the LLMStudio Web Interface

**Comment by rchard** [comment]
Thank you for the detailed update.  Not sure if this is appropriate place to ask, but when I go through these steps the application is able to start:

INFO:     Will watch for changes in these directories: ['/home/iroot/repo/h2o-llmstudio']
INFO:     Uvicorn running on http://127.0.0.1:8756 (Press CTRL+C to quit)
INFO:     Started reloader process [170661] using WatchFiles
Error loading keyring: No recommended backend was available. Install a recommended 3rd party backend package; or, install the keyrings.alt package if you want to use the non-recommended backends. See https://pypi.org/project/keyring for details.. Disabling keyring save option.
INFO:     Started server process [170669]
INFO:     Waiting for application startup.
2023-09-14 11:50:32,745 - INFO: STARTING APP
INFO:     Application startup complete.
2023/09/14 11:50:36 # {"addr":"[::1]:36242","route":"/","t":"ui_add"}
2023/09/14 11:50:38 # {"addr":"[::1]:36242","t":"ui_drop"}
2023/09/14 11:50:38 # {"addr":"[::1]:36250","route":"/","t":"ui_add"}

...but I am unable get anything from localhost:10101 other than the blinking WAVE LIVE icon.  Any advice would be welcome (including a "please go here to ask this question" :) .  Thanks.

**Comment by pascal-pfeiffer** [comment]
Thank you for reporting the issue @rchard .
As I couldn't reproduce it locally, could you please enable the Wave server logs and send us the full log:
basically, remove `H2O_WAVE_NO_LOG=True \` from the make command or set to `False`

**Comment by rchard** [comment]
Very new to this platform so forgive the obvious questions.  Where would the logs be:

(h2otune) iroot@jf4chatpde02:~/repo/h2o-llmstudio$ H2O_WAVE_APP_ADDRESS=http://127.0.0.1:8756 H2O_WAVE_MAX_REQUEST_SIZE=25MB H2O_WAVE_NO_LOG=False H2O_WAVE_PRIVATE_DIR="/download/@output/download" wave run app
2023/09/15 10:09:15 #
2023/09/15 10:09:15 # ┌────────────────┐ H2O Wave
2023/09/15 10:09:15 # │  ┐┌┐┐┌─┐┌ ┌┌─┐ │ 0.26.2 20230803101404
2023/09/15 10:09:15 # │  └┘└┘└─└└─┘└── │ © 2021 H2O.ai, Inc.
2023/09/15 10:09:15 # └────────────────┘
2023/09/15 10:09:15 # ┌──────────────────────────────────────┐
2023/09/15 10:09:15 # │  Running at http://localhost:10101/  │
2023/09/15 10:09:15 # └──────────────────────────────────────┘
2023/09/15 10:09:15 # {"address":"download/","source":"output/download","t":"private_dir"}
2023/09/15 10:09:15 # {"address":":10101","base-url":"/","t":"listen","web-dir":"/home/iroot/miniconda3/envs/h2otune/www"}
INFO:     Will watch for changes in these directories: ['/home/iroot/repo/h2o-llmstudio']
INFO:     Uvicorn running on http://127.0.0.1:8756 (Press CTRL+C to quit)
INFO:     Started reloader process [359777] using WatchFiles
Error loading keyring: No recommended backend was available. Install a recommended 3rd party backend package; or, install the keyrings.alt package if you want to use the non-recommended backends. See https://pypi.org/project/keyring for details.. Disabling keyring save option.
INFO:     Started server process [359785]
INFO:     Waiting for application startup.
2023-09-15 10:09:23,825 - INFO: STARTING APP
INFO:     Application startup complete.

**Comment by pascal-pfeiffer** [comment]
The logs will be printed to the console just like what you already are seeing. Could you try to connect to that server again now by browsing to `http://localhost:10101/` and paste the log. I am wondering if the wave server gives any additional error other than `ui_drop`

**Comment by rchard** [comment]
After a load and a couple of reloads:

(h2otune) iroot@jf4chatpde02:~/repo/h2o-llmstudio$ H2O_WAVE_APP_ADDRESS=http://127.0.0.1:8756 H2O_WAVE_MAX_REQUEST_SIZE=25MB H2O_WAVE_NO_LOG=False H2O_WAVE_PRIVATE_DIR="/download/@output/download" wave run app
2023/09/18 11:57:04 #
2023/09/18 11:57:04 # ┌────────────────┐ H2O Wave
2023/09/18 11:57:04 # │  ┐┌┐┐┌─┐┌ ┌┌─┐ │ 0.26.2 20230803101404
2023/09/18 11:57:04 # │  └┘└┘└─└└─┘└── │ © 2021 H2O.ai, Inc.
2023/09/18 11:57:04 # └────────────────┘
2023/09/18 11:57:04 # ┌──────────────────────────────────────┐
2023/09/18 11:57:04 # │  Running at http://localhost:10101/  │
2023/09/18 11:57:04 # └──────────────────────────────────────┘
2023/09/18 11:57:04 # {"address":"download/","source":"output/download","t":"private_dir"}
2023/09/18 11:57:04 # {"address":":10101","base-url":"/","t":"listen","web-dir":"/home/iroot/miniconda3/envs/h2otune/www"}
INFO:     Will watch for changes in these directories: ['/home/iroot/repo/h2o-llmstudio']
ERROR:    [Errno 98] Address already in use
(h2otune) iroot@jf4chatpde02:~/repo/h2o-llmstudio$ 2023/09/18 11:57:19 # {"addr":"[::1]:37444","route":"/","t":"ui_add"}
^C
(h2otune) iroot@jf4chatpde02:~/repo/h2o-llmstudio$ H2O_WAVE_APP_ADDRESS=http://127.0.0.1:8756 H2O_WAVE_MAX_REQUEST_SIZE=25MB H2O_WAVE_NO_LOG=False H2O_WAVE_PRIVATE_DIR="/download/@output/download" wave run app
INFO:     Will watch for changes in these directories: ['/home/iroot/repo/h2o-llmstudio']
INFO:     Uvicorn running on http://127.0.0.1:8756 (Press CTRL+C to quit)
INFO:     Started reloader process [352329] using WatchFiles
Error loading keyring: No recommended backend was available. Install a recommended 3rd party backend package; or, install the keyrings.alt package if you want to use the non-recommended backends. See https://pypi.org/project/keyring for details.. Disabling keyring save option.
INFO:     Started server process [352331]
INFO:     Waiting for application startup.
2023-09-18 11:58:23,605 - INFO: STARTING APP
INFO:     Application startup complete.
2023/09/18 11:58:45 # {"addr":"[::1]:37444","t":"ui_drop"}
2023/09/18 11:58:45 # {"addr":"[::1]:52552","route":"/","t":"ui_add"}
2023/09/18 11:59:09 # {"addr":"[::1]:52552","t":"ui_drop"}
2023/09/18 11:59:09 # {"addr":"[::1]:47018","route":"/","t":"ui_add"}

**Comment by pascal-pfeiffer** [comment]
Thank you. In your case the proxy seems to be the issue. With the proxy set, the app never connects to the wave server (`app_add` is missing from the logs)

What exactly is your proxy needed for?
Were you able to start the server now without the proxy?

**Comment by rchard** [comment]
No, it just hangs at "Downloading default dataset"

**Comment by pascal-pfeiffer** [comment]
I see. So, what H2O LLM Studio does in the background is

```python
from datasets import load_dataset
ds = load_dataset("OpenAssistant/oasst1")
```

Please set up your machine that this succeeds to be able to use the default dataset. 
If for any reasons, huggingface server must be blocked, you can also download elsewhere and manually add the datase to you local huggingface cache (by default, located at `.cache/huggingface/datasets/`)

**Comment by rchard** [comment]
First off, thank you for your replies you've given me hope but I'm not quite there yet!  I did have to download the datasets module to my environment.  When I run the code snippet that you mentioned above, it hangs unless I set the environment variable http_proxy.  Then it downloads the data set with no issue.  Unfortunately I still get the same "Loading..." prompt.  Is "OpenAssistant/oasst1" the only set it's looking for?  As I pointed to earlier, if i just set the environment variable and then run wave, it hangs earlier and doesn't even get to the "Loading..." prompt.  

If "OpenAssistant/oasst1" isn't the only dataset it needs is there any way getting a list of what it needs downloaded and doing that manually?  Do I need to do anything to the run command if I go this route (something like an --ofline switch)?  thanks again for your help!
